server.js

```diff
@@ -17,20 +18,21 @@ const { PORT = 4000 } = process.env;
 
 // Add compression logic and handle compressed files
 app.use(compression());
-app.use('/', expressStaticGzip(appPath, {
-    enableBrotli: true,
-    orderPreference: ['br', 'gz'],
-}));
-app.use('/images', express.static(appPath));
-
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
 
 /* Cross-Origin Resource Sharing CORS
 To enable access of resources from our server */
 isDevEnv && app.use(cors());
 
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+
 app.use('/', router);
+app.use('/', expressStaticGzip(appPath, {
+    enableBrotli: true,
+    orderPreference: ['br', 'gz'],
+}));
+
+app.get('*', (req, res) => res.sendFile(path.resolve(appPath, 'index.html')));
```

Comment:
how does this fixed the issue?

Reply:
It is saying, take every request that is left unhandled (if you look in the code above `line 35` we are handling the many request to the app `/` and the routes `/`) and send it back to the app (`index/html`). Once we send `index/html` react automatically reattaches itself to the `DOM` and then `react-router` handles appropriately routing the app to the current url path.